artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/AMQPConnectionContext.java

```diff
@@ -191,17 +190,13 @@ public void flush() {
    }
 
    public void close(ErrorCondition errorCondition) {
-      synchronized (schedulingLock) {
-         isSchedulingCancelled = true;
-
-         if (scheduledPool != null && scheduledPool instanceof ThreadPoolExecutor &&
-            scheduledFuture != null && scheduledFuture instanceof Runnable) {
-            if (!((ThreadPoolExecutor) scheduledPool).remove((Runnable) scheduledFuture) &&
-               !scheduledFuture.isCancelled() && !scheduledFuture.isDone()) {
-               log.warn("Scheduled task can't be removed from scheduledPool.");
-            } else {
-               scheduledFuture = null;
-            }
+      Future scheduledFuture = scheduledFutureRef.getAndSet(null);
+
+      if (scheduledPool != null && scheduledPool instanceof ThreadPoolExecutor &&
+         scheduledFuture != null && scheduledFuture != voidFuture && scheduledFuture instanceof Runnable) {
```

Comment:
scheduledFuture != null is redundent, due to scheduledFuture instanceOf runnable check.